staging/src/k8s.io/apiserver/pkg/server/handler.go

```diff
@@ -48,14 +49,28 @@ type APIServerHandler struct {
 	// PostGoRestfulMux is the final HTTP handler in the chain.
 	// It comes after all filters and the API handling
 	PostGoRestfulMux *mux.PathRecorderMux
+
+	// Director is here so that we can properly handle fall through and proxy cases.
```

Comment:
let me explain. no, there is too much. let me sum up.

explaining the data flow of the four handlers (and maybe tweaking names) would help enormously:
```
FullHandlerChain -> Director -> {GoRestfulContainer,PostGoRestfulMux} based on inspection of registered web services
GoRestfulContainer -> {web services, PostGoRestfulMux}
```

also, what each should be used for externally, and who is responsible for building each